Freeze when 'singleplayer' button is clicked with only fabric-item-api-v1

Environment:
Prism Launcher 9.4
Minecraft 1.21.11
Fabric Loader 0.18.2
fabric-item-api-v1-11.5.19+78c8b46653.jar
fabric-api-base-1.0.5+4ebb5c0853.jar

STR:
1. Launch Minecraft
2. Click continue
3. Click singleplayer

Actual behaviour:
Freezes and I had to close the game. 

The issue does not occur when I add fabric-resource-loader-v1 and its dependency fabric-resource-loader-v0

Logs:
[latest.log](https://github.com/user-attachments/files/24070056/latest.log)

Excerpt of logs
<details><summary>Details</summary>

```
Error executing task on Client
net.minecraft.class_148: mouseClicked event handler
	at knot//net.minecraft.class_312.method_1601(class_312.java:133)
	at knot//net.minecraft.class_312.method_22686(class_312.java:244)
	at knot//net.minecraft.class_1255.method_18859(class_1255.java:169)
	at knot//net.minecraft.class_4093.method_18859(class_4093.java:23)
	at knot//net.minecraft.class_1255.execute(class_1255.java:111)
	at knot//net.minecraft.class_312.method_22684(class_312.java:244)
	at knot//org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43)
	at knot//org.lwjgl.system.JNI.invokeV(Native Method)
	at knot//org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3438)
	at knot//com.mojang.blaze3d.systems.RenderSystem.pollEvents(RenderSystem.java:125)
	at knot//com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:136)
	at knot//net.minecraft.class_1041.method_15998(class_1041.java:313)
	at knot//net.minecraft.class_310.method_1523(class_310.java:1411)
	at knot//net.minecraft.class_310.method_1514(class_310.java:966)
	at knot//net.minecraft.client.main.Main.main(Main.java:250)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:514)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:72)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:105)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.NoSuchMethodError: 'net.minecraft.class_5352 net.minecraft.class_3298.getFabricPackSource()'
	at knot//net.fabricmc.fabric.impl.item.EnchantmentUtil.determineSource(EnchantmentUtil.java:80)
	at knot//net.minecraft.class_7655.wrapOperation$zzm000$fabric-item-api-v1$enchantmentKey(class_7655.java:563)
	at knot//net.minecraft.class_7655.method_56915(class_7655.java:312)
	at knot//net.minecraft.class_7655.method_45122(class_7655.java:327)
	at knot//net.minecraft.class_7655$class_9158.method_56520(class_7655.java:121)
	at knot//net.minecraft.class_7655.method_56514(class_7655.java:201)
	at knot//net.minecraft.class_7655.method_45120(class_7655.java:220)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at knot//net.minecraft.class_7655.method_45121(class_7655.java:220)
	at knot//net.minecraft.class_7655.method_56515(class_7655.java:201)
	at knot//net.minecraft.class_7237.method_42098(class_7237.java:42)
	at knot//net.minecraft.class_525.method_64244(class_525.java:358)
	at knot//net.minecraft.class_525.method_64368(class_525.java:318)
	at knot//net.minecraft.class_525.method_31130(class_525.java:305)
	at knot//net.minecraft.class_528.method_44678(class_528.java:173)
	at knot//net.minecraft.class_528.<init>(class_528.java:126)
	at knot//net.minecraft.class_528$class_11780.method_73460(class_528.java:742)
	at knot//net.minecraft.class_526.method_25426(class_526.java:75)
	at knot//net.minecraft.class_437.method_25423(class_437.java:326)
	at knot//net.minecraft.class_310.method_1507(class_310.java:1237)
	at knot//net.minecraft.class_442.method_19861(class_442.java:148)
	at knot//net.minecraft.class_4185.method_25306(class_4185.java:109)
	at knot//net.minecraft.class_4264.method_25348(class_4264.java:53)
	at knot//net.minecraft.class_339.method_25402(class_339.java:122)
	at knot//net.minecraft.class_4069.method_25402(class_4069.java:46)
	at knot//net.minecraft.class_442.method_25402(class_442.java:277)
	at knot//net.minecraft.class_312.method_1601(class_312.java:122)
	... 20 more
``` 

</details> 

---

I also noticed a crash with fabric-item-api-v1 on 1.21.10. Same steps to reproduce. 

[latest.log](https://github.com/user-attachments/files/24070102/latest.log)
[crash-2025-12-10_16.43.01-client.txt](https://github.com/user-attachments/files/24070104/crash-2025-12-10_16.43.01-client.txt)
From the `java.lang.ClassNotFoundException: net.fabricmc.fabric.impl.resource.loader.FabricResource` I tried adding fabric-resource-loader-v1 and it worked fine.

--- a/fabric-item-api-v1/src/main/resources/fabric.mod.json
+++ b/fabric-item-api-v1/src/main/resources/fabric.mod.json
@@ -24,7 +24,8 @@
   ],
   "depends": {
     "fabricloader": ">=0.17.3",
-    "fabric-api-base": "*"
+    "fabric-api-base": "*",
+    "fabric-resource-loader-v1": "*"
   },
   "description": "Hooks for items",
   "custom": {

